### PR TITLE
Make dashpole a codeowner of prometheus exporters

### PR DESCRIPTION
**Pull request by dashpole**
**Description:**

Add myself as a codeowner of PRW and prometheus exporter components, and the PRW translation library.  I have some additional capacity to take these on, and will help with upcoming prometheus features (creation time, PRW 2.0, etc).

This will help me facilitate the LFX internship i'm co-mentoring: https://mentorship.lfx.linuxfoundation.org/project/3fa26f90-87aa-46a4-80f9-00195ae276e8

cc @Aneurysm9 @rapphil @ArthurSens @aknuds1

### CODE REVIEW

**Comment by rapphil** [comment]
I support @dashpole being a code owner of the prometheusremotewrite exporter.

**Review by rapphil — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Review by crobert-1 — COMMENTED** [review]
It's actually a bit more complicated to update code owners now, the `metadata.yaml` file of every component needs updated to include your username. Then some `make` commands need run, [here's the more in-depth details](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#how-to-become-a-code-owner). 👍 

Thanks for being willing to help out here @dashpole, really appreciate it!

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -71,8 +71,8 @@ exporter/mezmoexporter/                                             @open-teleme
 exporter/opencensusexporter/                                        @open-telemetry/collector-contrib-approvers @open-telemetry/collector-approvers
 exporter/opensearchexporter/                                        @open-telemetry/collector-contrib-approvers @Aneurysm9 @MitchellGale @MaxKsyunz @YANG-DB
 exporter/otelarrowexporter/                                         @open-telemetry/collector-contrib-approvers @jmacd @moh-osman3 @lquerel
-exporter/prometheusexporter/                                        @open-telemetry/collector-contrib-approvers @Aneurysm9
-exporter/prometheusremotewriteexporter/                             @open-telemetry/collector-contrib-approvers @Aneurysm9 @rapphil
+exporter/prometheusexporter/                                        @open-telemetry/collector-contrib-approvers @Aneurysm9 @dashpole
```
Comment (songy23) [inline .github/CODEOWNERS:74]: I believe you need to update https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/prometheusexporter/metadata.yaml#L11 too, same for other components

Comment (dashpole) [reply .github/CODEOWNERS:74]: I believe I did that in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34795/commits/f1b2698e0bc8861e8f5a26fe72cc68b08dbf9e58

**Review by songy23 — APPROVED** [review]

**Review by crobert-1 — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
Thanks @dashpole!

**Pull request merged**